dhash/module.py

```diff
@@ -258,57 +301,56 @@ async def on_message_delete(self, message: discord.Message):
             break
 
     @commands.Cog.listener()
-    async def on_reaction_add(self, reaction, user):
+    async def on_raw_reaction_add(self, payload):
         """Handle 'This is a repost' report.
         The footer contains reposter's user ID and repost message id.
         """
-        if not self._in_repost_channel(reaction.message):
-            return
-        if user.bot:
+        hash_channel = HashChannel.get(payload.guild_id, payload.channel_id)
+
+        if not hash_channel:
             return
-        if not reaction.message.author.bot:
+        if payload.member.bot:
             return
-        emoji = str(reaction.emoji)
+
+        emoji = str(payload.emoji)
+
         if emoji != "❎":
             return
 
-        try:
-            repost_message_id = int(
-                reaction.message.embeds[0].footer.text.split(" | ")[1]
-            )
-            repost_message = await reaction.message.channel.fetch_message(
-                repost_message_id
-            )
-        except discord.errors.HTTPException as exc:
-            return await bot_log.error(
-                "Could not find repost message {msg_id} at guild {guild}".format(
-                    msg_id=reaction.message.id, guild=reaction.guild.id
-                ),
-                exception=exc,
-            )
+        channel = await self.bot.fetch_channel(payload.channel_id)
+        message = await channel.fetch_message(payload.message_id)
+
+        if not message.author.bot:
+            return
 
-        for report_reaction in reaction.message.reactions:
+        for report_reaction in message.reactions:
             if str(report_reaction) != "❎":
                 continue
 
-            if (
-                emoji == "❎"
-                and str(report_reaction) == "❎"
-                and report_reaction.count > NOT_DUPLICATE_LIMIT
-            ):
+            if report_reaction.count > hash_channel.reaction_limit:
                 # remove bot's reaction, it is not a repost
+
                 try:
+                    repost_message_id = int(
+                        message.embeds[0].footer.text.split(" | ")[1]
+                    )
+
+                    repost_message = await message.channel.fetch_message(
+                        repost_message_id
+                    )
+                    await repost_message.remove_reaction("♻️", self.bot.user)
+                    await repost_message.remove_reaction("♻️", self.bot.user)
                     await repost_message.remove_reaction("♻️", self.bot.user)
```

Comment:
You are removing the :recycle: reaction three times, instead of trying to remove the three types of them.